src/plugins/kibana/public/settings/sections/indices/add_data_steps/parse_csv_step/parse_csv_step.js

```diff
@@ -0,0 +1,79 @@
+import _ from 'lodash';
+import Papa from 'papaparse';
+import modules from 'ui/modules';
+import template from './parse_csv_step.html';
+import './styles/_add_data_parse_csv_step.less';
+
+modules.get('apps/settings')
+  .directive('parseCsvStep', function () {
+    return {
+      restrict: 'E',
+      template: template,
+      scope: {
+        file: '=',
+        parseOptions: '=',
+        samples: '='
+      },
+      bindToController: true,
+      controllerAs: 'wizard',
+      controller: function ($scope) {
+
+        this.delimiterOptions = [
+          {
+            label: 'comma',
+            value: ','
+          },
+          {
+            label: 'tab',
+            value: '\t'
+          },
+          {
+            label: 'space',
+            value: ' '
+          },
+          {
+            label: 'semicolon',
+            value: ';'
+          },
+          {
+            label: 'pipe',
+            value: '|'
+          }
+        ];
+
+        this.parse = () => {
+          if (!this.file) return;
+
+          const config = _.assign(
+            {
+              header: true,
+              preview: 10,
+              dynamicTyping: true,
+              complete: (results) => {
+                $scope.$apply(() => {
+                  this.formattedErrors = _.map(results.errors, (error) => {
+                    return `${error.type} at row ${error.row} - ${error.message}`;
+                  });
+                  this.columns = results.meta.fields;
+                  this.rows = _.map(results.data, _.values);
```

Comment:
Is this a potential bug? 
From the lodash documentation
> ```_.values(new Foo);
// → [1, 2] (iteration order is not guaranteed)```

Wouldn't this imply that the order of the fields that are being mapped could be in a different order from row to row in the results?